llvm/include/llvm/IR/IntrinsicInst.h

```diff
@@ -134,8 +134,14 @@ class IntrinsicInst : public CallInst {
       return CF->isIntrinsic();
     return false;
   }
+  static bool classof(const CallBase *I) {
+    if (const Function *CF = I->getCalledFunction())
+      return CF->isIntrinsic();
+    return false;
+  }
   static bool classof(const Value *V) {
-    return isa<CallInst>(V) && classof(cast<CallInst>(V));
+    return (isa<CallInst>(V) && classof(cast<CallInst>(V))) ||
+           (isa<CallBase>(V) && classof(cast<CallBase>(V)));
```

Comment:
I'm not sure if we should let these be treated as an ordinary IntrinsicInst. If we do, this should be restricted to just the CallInst + CallBr cases. Can this be left for later?

Reply:
In the current state of the code this is needed because, otherwise, converting the seldag/gisel functions to handle CallBase instead of CallInst doesn't work. (If we don't do this whole conversion, however, (see comments above) this classof adaption is no longer required, of course.)